_posts/asyncapi-tool-in-our-toolbox.md

```diff
@@ -0,0 +1,94 @@
+
+## Services share schema and contract, not class
+
+What publishers share with subscribers using NuGet packages, for example, are schemas and contracts. By chance, we respect the SOA third tenet using C# classes or interfaces to define the mentioned contracts. However, despite being indeed comfortable, that's not a requirement.
+
+[AsyncAPI](https://www.asyncapi.com) is a protocol-agnostic specification designed to describe asynchronous APIs:
+
+```yaml
+asyncapi: 2.2.0
+info:
+  title: Account Service
+  version: 1.0.0
+  description: This service is in charge of processing user signups
+channels:
+  user/signedup:
+    subscribe:
+      message:
+        $ref:'#/components/messages/UserSignedUp'
+components:
+  messages:
+    UserSignedUp:
+      payload:
+        type: object
+        properties:
+          displayName:
+            type: string
+            description: Name of the user
+          email:
+            type: string
+            format: email
+            description: Email of the user
+```
+
+(from AsyncAPI's samples)
+
+The presented YAML document describes the `UserSignedUp` event much better than what a NuGet package with a C# class and accompanying documentation can do. If not only, for the simple fact that everything is in the same place, the document.
+
+The second interesting aspect of the AsyncAPI specification is the community-driven set of generators. For example, there is a [template to generate C# code from an AsyncAPI YAML document](https://github.com/jonaslagoni/asyncapi-quicktype-template). There are also generators to output HTML documents for a documentation website.
+
+The fact that all information is in a single place, the YAML document, and that there are generators available makes it so that teams can share AsyncAPI specs documents instead of C# compiled artifacts. Once consumers and producers reference the YAML files, they can apply the templates to transform them into C# classes or interfaces.
+
+The advantage of this approach is that it entirely removes platform coupling from the equation. The system can be polyglot. Each endpoint can be coded using the platform and the language of choice.
+
+## Conclusion
+
+In many cases, we're sharing NuGet packages and .NET assemblies when we need to share message contracts because that's the simplest thing we can do. Not to mention that it's also a scenario well supported by the toolchain in use, for example, Visual Studio or JetBrains Rider. There are scenarios in which the limitations imposed by the generated platform coupling are a blocker. AsyncAPI might come to the rescue and save our day.

```

Comment:
```suggestion
In many cases, we're sharing NuGet packages and .NET assemblies when we need to share message contracts because that's the simplest thing we can do. Not to mention that it's also a scenario well supported by the toolchain in use, for example, Visual Studio or JetBrains Rider. There are scenarios in which the limitations imposed by the generated platform coupling are a blocker. AsyncAPI might come to the rescue and save the day.

```